### PR TITLE
Add Transforms Benchmark

### PR DESCRIPTION
**Pull request by isaaccorley**
- Add `benchmark_transforms.py` which allows one to compare the execution time of a sequential pipeline of our transforms + kornia augmentations for CPU vs GPU and various batch image/mask sizes.

### CODE REVIEW

**Comment by isaaccorley** [comment]
> Overall thoughts:
> 
> 1. How fast are these transforms? I assume ~1 sec or less. We might want to loop over them multiple times to get a more accurate speed measurement.
> 2. Do we want to benchmark the CPU transforms in parallel with multiple workers? Users could use them after data loading or as an argument to `dataset` right?

1. Depends. CPU/GPU are quite similar for small images and small batch sizes, as expected. For the opposite case, GPU will outperform CPU greatly. However it's not exactly an apples to apples comparison as you mention because CPU, in this case, is not taking advantage of any parallelism.
2. No we don't. This is a tricky one to measure but I think we can get use something like a `torch.utils.data.TensorDataset` so there should be no overhead for I/O since the tensors will already be loaded to memory.

**Comment by isaaccorley** [comment]
Also, other benchmark scripts and experiment scripts are missing the license comment, just a heads up.

**Comment by edgarriba** [comment]
> Let's not merge this yet so I can implement a true comparison per my above comment for GPU augs vs CPU augs with multiple workers.

let us know about this and see if we can be of any help here. We are currently refactoring the augmentations module just reorganizing code for further improvements /cc @shijianjian @twsl

**Comment by adamjstewart** [comment]
@isaaccorley is this still a WIP or should we close it?

**Comment by isaaccorley** [comment]
This is still a WIP but it's close to done. I think the only thing that's left is to figure out how we want to compare to CPU in an apples to apples way.
